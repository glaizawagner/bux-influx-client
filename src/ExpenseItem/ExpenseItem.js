/* eslint-disable no-unused-expressions */
import React from 'react';
import BuxInfluxContext from '../BuxInfluxContext';
// import PropTypes from 'prop-types';
import config from '../config'

function deleteExpenseRequest(eid, cb) {
    fetch(config.API_ENDPOINT +`/expenses/${eid}`, {
        method: 'DELETE',
        headers: {
            'content-type': 'application/json',
            // 'authorization': `bearer ${config.API_KEY}`
        }
    })
        .then(res => {
            if (!res.ok) {
            return res.json().then(error => Promise.reject(error))
            }
            // return res.json()
        })
        .then(() => {
            console.log(`For Eid: ${eid}`)
            cb(eid)
        })
        .catch(error => {
            console.error(error)
        })
}

export default function ExpenseItem(props) {
    return(
        <BuxInfluxContext.Consumer>
            { (context) => (
                <li className ='ExpenseItem'>
                    <div className = 'ExpenseItem__row'>
                        <span> 
                            {props.date_created} 
                            {props.description} 
                            {props.value} 
                            <button 
                                className='ExpenseItem__btn'
                                onClick={() => {
                                    deleteExpenseRequest(
                                        props.eid,
                                        context.deleteExpenses
                                    )
                                }}
                            >
                            Delete
                            </button>
                        </span>
                    </div>
                </li>
            )}
        </BuxInfluxContext.Consumer>
    )
}

