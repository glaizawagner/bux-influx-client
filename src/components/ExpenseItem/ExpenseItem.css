.ExpenseItem__row {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-self: flext-start;
    /* margin: 0 auto; */
    padding: 5px 0;
    font-size: 1.5rem;
}

.ExpenseItem__btn {
    font-size: 1.5rem;
    background: transparent;
    border: 0;
}

.ExpenseItem__btn:hover {
   font-weight: 900;
    color: blue;
    transform: translateY(-2px);
    cursor: pointer;
}

.timesIcon {
    font-size: 2rem;
    border: transparent;
    margin-top: 2px
}

.items {
    margin: 0 1rem;
} 

.exp-extra {
    color: #ff5033;
} 


@media only screen and (min-width: 37.5em) and (max-width: 56.25em) {
    .ExpenseItem__row {
        margin: 0 1rem;
    }
}

@media only screen and (min-width: 56.25em) and (max-width: 75em) {
    .ExpenseItem__row {
        margin: 0 2rem;
    }
}

@media only screen and (min-width: 75em) and (max-width: 105em) {
    .ExpenseItem__row {
        margin: 0 2rem;
    }
}