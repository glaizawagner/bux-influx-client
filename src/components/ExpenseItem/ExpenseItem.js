/* eslint-disable no-unused-expressions */
import React from 'react';
import BuxinfluxContext from '../../contexts/BuxinfluxContext';
import BuxinfluxApiService from '../../services/buxinflux-api-service'
import helpers from '../../components/helpers/helpers';
// import PropTypes from 'prop-types';
// import config from '../../config'
import { Icon } from '@iconify/react';
import timesCircle from '@iconify/icons-fa-regular/times-circle';


function deleteExpenseRequest(eid,cb) {
    BuxinfluxApiService.deleteExpenses(eid)
    .then(() => {
        cb(eid)
    })
    .catch((e) => console.error(e));
}

export default function ExpenseItem(props) {
    return(
        <BuxinfluxContext.Consumer>
            { (context) => (
                <li className ='ExpenseItem'>
                    <div className = 'ExpenseItem__row'>
                        <span > 
                        <span className="items">{helpers.formatDateDisplay(props.date_created)} </span>
                        <span className="items">{props.description} </span>
                        <span className="items">{props.value} </span>
                        <span className="items">{props.percentage} </span>
                            <button 
                                className='ExpenseItem__btn'
                                onClick={() => {
                                    deleteExpenseRequest(
                                        props.eid,
                                        context.deleteExpenses
                                    )
                                }}
                            >
                            <Icon icon={timesCircle} className="timesIcon" />
                            </button>
                        </span>
                    </div>
                </li>
            )}
        </BuxinfluxContext.Consumer>
    )
}

ExpenseItem.defaultProps = {
    deleteExpenses: () => {}
}
