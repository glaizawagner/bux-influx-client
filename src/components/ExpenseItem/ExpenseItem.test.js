import React from 'react';
import { shallow } from 'enzyme';
import ExpenseItem from '../ExpenseItem/ExpenseItem';
import toJson from 'enzyme-to-json'

describe('Expense Item  component', () => {
    const props = {
        eid: 1,
        date_created: '2019-11-12T16:28:32.615Z',
        type: 'exp',
        description: 'Salary',
        value: '3500.00',
        percentage: '17.00',
        user_id: 1
    };
    
    it('renders Expenses by default', () => {
        const wrapper = shallow(<ExpenseItem {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders Expense given props', () => {
        const wrapper = shallow(<ExpenseItem />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});