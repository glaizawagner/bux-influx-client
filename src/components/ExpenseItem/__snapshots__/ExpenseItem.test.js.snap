// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense Item  component renders Expense given props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Expense Item  component renders Expenses by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
